python/lsst/ip/isr/linearize.py

```diff
@@ -127,32 +139,280 @@ def fromYaml(self, yamlObject):
+
+            linDict['amplifiers'][ampName] = ampDict
+
+        try:
+            table = afwTable.BaseCatalog.readFits(filename, 2)
+            tableData = []
+            for row in table:
+                record = row.extract('*')
+                tableData.append(record['LOOKUP_VALUES'])
+            linDict['tableData'] = tableData
+        except Exception:
+            pass
+
+        return cls().fromYaml(linDict)
+
+    def writeFits(self, filename):
+        """Write the linearity model to a FITS file.
+
+        Parameters
+        ----------
+        filename : `str`
+            Name of the file to write.
+
+        Returns
+        -------
+        used : `str`
+            The name of the file used to write the data.
+
+        Notes
+        -----
+        The file is written to YAML format and will include any metadata
+        associated with the `Linearity`.
+        """
+        now = datetime.datetime.utcnow()
+        self.updateMetadata(date=now)
+
+        metadata = copy.copy(self.getMetadata())
+        metadata["LINEARITY_SCHEMA"] = "Linearity table"
+        metadata["LINEARITY_VERSION"] = 1
+
+        # Now pack it into a fits table.
+        length = max([len(self.linearityCoeffs[x]) for x in self.linearityCoeffs.keys()])
+
+        schema = afwTable.Schema()
+        names = schema.addField("AMPLIFIER_NAME", type="String", size=16, doc="linearity amplifier name")
+        types = schema.addField("TYPE", type="String", size=16, doc="linearity type names")
+        coeffs = schema.addField("COEFFS", type="ArrayD", size=length, doc="linearity coefficients")
+        boxX = schema.addField("BBOX_X0", type="I", doc="linearity bbox minimum x")
+        boxY = schema.addField("BBOX_Y0", type="I", doc="linearity bbox minimum y")
+        boxDx = schema.addField("BBOX_DX", type="I", doc="linearity bbox x dimension")
+        boxDy = schema.addField("BBOX_DY", type="I", doc="linearity bbox y dimension")
+
+        catalog = afwTable.BaseCatalog(schema)
+        catalog.resize(len(self.linearityCoeffs.keys()))
+
+        for ii, ampName in enumerate(self.linearityType):
+            catalog[ii][names] = ampName
+            catalog[ii][types] = self.linearityType[ampName]
+            catalog[ii][coeffs] = self.linearityCoeffs[ampName]
+
+            bbox = self.linearityBBox[ampName]
+            catalog[ii][boxX], catalog[ii][boxY] = bbox.getMin()
+            catalog[ii][boxDx], catalog[ii][boxDy] = bbox.getDimensions()
+        catalog.setMetadata(metadata)
+        catalog.writeFits(filename)
+
+        if self.tableData is not None:
+            schema = afwTable.Schema()
+            dimensions = self.tableData.shape
+            lut = schema.addField("LOOKUP_VALUES", type='ArrayI', size=dimensions[1],
+                                  doc="linearity lookup data")
+            catalog = afwTable.BaseCatalog(schema)
+            catalog.resize(dimensions[0])
+
+            for ii in range(dimensions[0]):
```

Comment:
afw allows you to assign numpy arrays directly to a column so given you have one column you might be able to assign the entire tableData array to the afwTable in one assignment. @TallJimbo will this work?

Reply:
Can't assign multiple columns at once, I'm afraid.

But to back up a bit - why are we using `afw.table` at all here (or in defects, for that matter)?  Do we use this data structure in C++ at some point?

Reply:
No. We used it to so that we can call Catalog.writeFits with a PropertyList and afwTable. If you want us to only use astropy.table to write and read fits tables we can switch but that won't be immediate.

Note that we are assigning a single column here but it's a 1D cell.

Reply:
I think I would recommend astropy in the longer term; I'm trying to reduce our usage of afw.table when there's something third-party that does the job, and while that has focused on Pandas/Parquet for catalogs, it probably should apply to astropy for general FITS I/O as well.

(Or something else - I know several DM members who prefer Erin Sheldon's fitsio package, as it's a lot faster, but for a lot of stuff even astropy is faster than afw.table.)